scripts/feature_tests/upgrade/upgrade_common.sh

```diff
@@ -0,0 +1,448 @@
+    export NUM_OF_MASTER_REPLICAS="${1}"
+}
+
+function set_number_of_worker_node_replicas() {
+    export NUM_OF_WORKER_REPLICAS="${1}"
+}
+
+function provision_controlplane_node() {
+    pushd "${METAL3_DEV_ENV_DIR}" || exit
+    echo "Provisioning a controlplane node...."
+    bash ./scripts/v1alphaX/provision_cluster.sh
+    bash ./scripts/v1alphaX/provision_controlplane.sh
+    popd || exit
+}
+
+function provision_worker_node() {
+    pushd "${METAL3_DEV_ENV_DIR}" || exit
+    echo "Provisioning a worker node...."
+    bash ./scripts/v1alphaX/provision_worker.sh
+    popd || exit
+}
+
+function deprovision_cluster() {
+    pushd "${METAL3_DEV_ENV_DIR}" || exit
+    echo "Deprovisioning the cluster...."
+    bash ./scripts/v1alphaX/deprovision_cluster.sh
+    popd || exit
+}
+
+function wait_for_cluster_deprovisioned() {
+    echo "Waiting for cluster to be deprovisioned"
+    for i in {1..3600}; do
+        cluster_count=$(kubectl get clusters -n metal3 2>/dev/null | awk 'NR>1' | wc -l)
+        if [[ "${cluster_count}" -eq "0" ]]; then
+            ready_bmhs=$(kubectl get bmh -n metal3 | awk 'NR>1' | grep -c 'ready')
+            if [[ "${ready_bmhs}" -eq "${NUM_NODES}" ]]; then
+                echo ''
+                echo "Successfully deprovisioned the cluster"
+                break
+            fi
+        else
+            echo -n "-"
+        fi
+        sleep 10
+    done
+}
+
+function deploy_workload_on_workers() {
+    echo "Deploy workloads on workers"
+    # Deploy workloads
+cat <<EOF | kubectl apply -f -
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: workload-1-deployment
+spec:
+  replicas: 10
+  selector:
+    matchLabels:
+      app: workload-1
+  template:
+    metadata:
+      labels:
+        app: workload-1
+    spec:
+      containers:
+      - name: nginx
+        image: nginx
+EOF
+
+    echo "Waiting for workloads to be ready"
+    for i in {1..1800}; do
+        workload_replicas=$(kubectl get deployments workload-1-deployment \
+            -o json | jq '.status.readyReplicas')
+        if [[ "$workload_replicas" == "10" ]]; then
+            echo ''
+            echo "Successfully deployed workloads across the cluster"
+            break
+        fi
+        echo -n "*"
+        sleep 10
+        if [[ "${i}" -ge 1800 ]]; then
+            log_error " Workload failed to be deployed on the cluster"
+            deprovision_cluster
+            wait_for_cluster_deprovisioned
+            break
+        fi
+    done
+}
+
+function manage_node_taints() {
+    # Enable workload on masters
+    ssh -o PasswordAuthentication=no -o "StrictHostKeyChecking no" \
+        "${IMAGE_USERNAME}@${1}" -- sudo apt-get install jq -y
+    # untaint all masters (one workers also gets untainted, doesn't matter):
+    ssh -o PasswordAuthentication=no -o "StrictHostKeyChecking no" \
+        "${IMAGE_USERNAME}@${1}" -- kubectl taint nodes --all node-role.kubernetes.io/master-
```

Comment:
this does not need to be done by ssh. you can get the kubeconfig from the management cluster. Check Feruz's PR https://github.com/metal3-io/metal3-dev-env/pull/385

Reply:
done